### PR TITLE
Utilities: Add a Brainf**k interpreter

### PR DESCRIPTION
**Pull request by ninadsachania**

### CODE REVIEW

File: Userland/Utilities/bf.cpp [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+    Core::ArgsParser args_parser;
+    args_parser.set_general_help("A Brainf**k interpreter.");
+    args_parser.add_positional_argument(path, "Program path", "path", Core::ArgsParser::Required::No);
+    args_parser.parse(arguments);
+
+    auto file_or_error = TRY(Core::File::open_file_or_standard_stream(path, Core::File::OpenMode::Read));
+
+    auto const content = TRY(file_or_error->read_until_eof());
+    StringView const content_view(content);
```
Comment (timschumi) [inline Userland/Utilities/bf.cpp:106]: This conversion can be done implicitly as far as I know.

File: Userland/Utilities/bf.cpp [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+#include <AK/StringView.h>
+#include <LibCore/ArgsParser.h>
+#include <LibCore/File.h>
+#include <LibCore/System.h>
+#include <LibMain/Main.h>
+
+static void execute(StringView const instructions)
+{
+    Array<u8, 30000> m_data;
```
Comment (timschumi) [inline Userland/Utilities/bf.cpp:15]: It might be useful to make this size configurable by a command line argument.

Comment (timschumi) [inline Userland/Utilities/bf.cpp:15]: None of the accesses to this array are bounds-checked. This implementation will either be trivial to crash or can be used in a "101" course on binary exploitation.

File: Userland/Utilities/bf.cpp [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    TRY(Core::System::pledge("stdio rpath"));
+
+    StringView path;
+
+    Core::ArgsParser args_parser;
+    args_parser.set_general_help("A Brainf**k interpreter.");
+    args_parser.add_positional_argument(path, "Program path", "path", Core::ArgsParser::Required::No);
+    args_parser.parse(arguments);
+
+    auto file_or_error = TRY(Core::File::open_file_or_standard_stream(path, Core::File::OpenMode::Read));
```
Comment (timschumi) [inline Userland/Utilities/bf.cpp:103]: This is no longer `_or_error` after `TRY`.

**Review by timschumi — CHANGES_REQUESTED** [review]
None of this is currently linked into the build system. Also, I'm generally unsure if this is a good fit for the system (or at least for the "utilities" section).

File: Userland/Utilities/bf.cpp (hunk 1 above)
Comment (ninadsachania) [reply Userland/Utilities/bf.cpp:106]: Thanks for the tip!

File: Userland/Utilities/bf.cpp (hunk 2 above)
Comment (ninadsachania) [reply Userland/Utilities/bf.cpp:15]: Done.

Comment (ninadsachania) [reply Userland/Utilities/bf.cpp:15]: I'll use a `Vector`. That'll be bounds-checked.

**Comment by ninadsachania** [comment]
> None of this is currently linked into the build system. Also, I'm generally unsure if this is a good fit for the system (or at least for the "utilities" section).

I think it'll be a fun little utility to explore a quite popular esolang. I am happy it to move it to a different place.

**Comment by nico** [comment]
I don't think we want utilities like this. It's not really useful for anything, it's not hard to write, there's no shortage of brainfuck interpreters on the Internet, and we're similarly no longer accepting games (which arguably have the benefit of demoing LibGUI) due to being low-effort. It's better to spend your energy improving the actual system 🙂

**Pull request closed without merging**